zkevm-circuits/src/evm_circuit/execution/codecopy.rs

```diff
@@ -207,217 +207,32 @@ impl<F: Field> ExecutionGadget<F> for CodeCopyGadget<F> {
-                rw_indices: vec![(RwTableTag::Stack, 2)],
-                execution_state: ExecutionState::PUSH,
-                rw_counter: 3,
-                program_counter: 66,
-                stack_pointer: 1022,
-                gas_left: total_gas_cost.sub(2 * gas_cost_push32),
-                gas_cost: gas_cost_push32,
-                opcode: Some(OpcodeId::PUSH32),
-                ..Default::default()
-            },
-            ExecStep {
-                rw_indices: vec![
-                    (RwTableTag::Stack, 3),
-                    (RwTableTag::Stack, 4),
-                    (RwTableTag::Stack, 5),
-                ],
-                execution_state: ExecutionState::CODECOPY,
-                rw_counter: 4,
-                program_counter: 99,
-                stack_pointer: 1021,
-                gas_left: gas_cost_codecopy,
-                gas_cost: gas_cost_codecopy,
-                opcode: Some(OpcodeId::CODECOPY),
-                ..Default::default()
-            },
-        ];
-
-        let program_counter = 100;
-        let stack_pointer = 1024;
-        let mut rw_counter = 7;
-        if !size.is_zero() {
-            make_copy_code_steps(
-                call_id,
-                &code,
-                src_addr,
-                dst_addr,
-                size,
-                program_counter,
-                stack_pointer,
-                next_memory_word_size * 32,
-                &mut rw_counter,
-                &mut rws_map,
-                &mut steps,
-            );
-        }
-        steps.push(ExecStep {
-            execution_state: ExecutionState::STOP,
-            rw_counter,
-            program_counter: 100,
-            stack_pointer: 1024,
-            opcode: Some(OpcodeId::STOP),
-            memory_size: next_memory_word_size * 32,
-            ..Default::default()
-        });
-
-        let block = Block {
-            randomness,
-            txs: vec![Transaction {
-                id: 1,
-                calls: vec![Call {
-                    id: call_id,
-                    is_root: true,
-                    is_create: false,
-                    code_source: CodeSource::Account(code.hash),
-                    ..Default::default()
-                }],
-                steps,
-                ..Default::default()
-            }],
-            rws: rws_map,
-            bytecodes: vec![code],
-            ..Default::default()
-        };
-        assert_eq!(run_test_circuit_incomplete_fixed_table(block), Ok(()));
-    }
-
-    #[test]
-    fn codecopy_gadget_single_step() {
-        test_ok(0x00, 0x00, 54);
-        test_ok(0x10, 0x05, 54);
-    }
-
-    #[test]
-    fn codecopy_gadget_multi_step() {
-        test_ok(0x00, 0x40, 123);
     }
 
     #[test]
-    fn codecopy_gadget_oob() {
-        test_ok(0x10, 0x20, 200);
+    fn codecopy_gadget() {
+        test_ok(0x00, 0x00, 0x20);
+        test_ok(0x20, 0x30, 0x30);
+        test_ok(0x10, 0x20, 0x42);
```

Comment:
Would be nice to be able to place code specifically at a place and with some values so that we can check if what is copied is what we're expecting specifically.

The ops match of course, so that's sign of that everything is correct. But further checking would be a nice-to-have

But I belive with the testing-setup we have now is not going to be feasible.

Reply:
Like the above pointer, it does make sense but I believe it would be better to track it separately :)